Libraries/ToTripleSlashPorter.cs

```diff
@@ -122,128 +134,196 @@ private void Port()
+                if (mainProjectInfo.Compilation.SyntaxTrees.FirstOrDefault(x => x.FilePath == tree.FilePath) is not null)
                 {
-                    // The symbol has to live in one of the current project's referenced projects
-                    foreach (ProjectReference projectReference in mainProjectData.Project.ProjectReferences)
-                    {
-                        PropertyInfo prop = typeof(ProjectId).GetProperty("DebugName", BindingFlags.NonPublic | BindingFlags.Instance)
-                            ?? throw new NullReferenceException("ProjectId.DebugName private property not found.");
+                    var symbolInfo = new SymbolInformation(docsType, mainProjectInfo);
+                    ResolvedSymbols.Add(symbol, symbolInfo);
+                }
+                // Then it should be located in one of the referenced projects
+                else
+                {
+                    FindSymbolInReferencedProjects(docsType, mainProjectInfo.Project.ProjectReferences);
+                }
+            }
 
-                        string projectPath = prop.GetValue(projectReference.ProjectId)?.ToString()
-                            ?? throw new NullReferenceException("ProjectId.DebugName value was null.");
+            PortResolvedSymbols();
+        }
 
-                        if (string.IsNullOrWhiteSpace(projectPath))
-                        {
-                            throw new Exception("Project path was empty.");
-                        }
+        private void FindSymbolInReferencedProjects(DocsType docsType, IEnumerable<ProjectReference> projectReferences)
+        {
+            foreach (ProjectReference projectReference in projectReferences)
+            {
+                string projectPath = GetProjectPath(projectReference);
+                string projectFileName = Path.GetFileNameWithoutExtension(projectPath);
 
-                        // Can't reuse the existing Workspace or exception thrown saying we already have the project loaded in this workspace.
-                        // Unfortunately, there is no way to retrieve a references project as a Project instance from the existing workspace.
-                        ProjectData extraProjectData = GetProjectDataAndSymbol(projectPath, docsType.FullName, out INamedTypeSymbol? actualSymbol);
 
-                        ResolvedSymbols.Add(actualSymbol, new SymbolData { Api = docsType, ProjectData = extraProjectData });
-                    }
+                // Can't reuse the existing Workspace or an exception is thrown saying we already have the project loaded in this workspace.
+                // Unfortunately, there is no way to retrieve a references project as a Project instance from the existing workspace.
+
+                ProjectInformation? pd = null;
+                INamedTypeSymbol? symbol = null;
+
+                if (TryFindSymbolInReferencedProject(projectPath, docsType.FullName, isMono: false, out pd, out symbol) ||
```

Comment:
This is not exactly what I meant. 

What I meant is if the symbol location is under `src/libraries/` tree. Even when the compilation doesn't set `RuntimeFlavor=mono`, it could resolve symbols where it's location is: `src/coreclr/System.Private.CoreLib/*`, and that is when the symbol would appear in the mono and coreclr System.Private.CoreLib.

Basically the way that you have it, this first call to `TryFindSymbolInReferenceProject` will always return true for `projectFileName == 'System.Private.CoreLIb`. The reason for that is because both System.Private.CoreLib need to have the same public apis defined.

Reply:
Ok thanks.
I need an example of this though. I can use an API from System.IO.FileSystem, like FileStream, to try to find a mono implementation. I think that type is one of those cases.

Reply:
@safern I think I am doing the right thing now. PTAL when you get a chance.